Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs

```diff
@@ -485,5 +490,207 @@ public static CairoSettingsWindow Instance
+                    break;
+                case "60": // Fit { WallpaperStyle = 6; TileWallpaper = 0 }
+                    style = Desktop.CairoWallpaperStyle.Fit;
+                    break;
+                case "100": // Fill { WallpaperStyle = 10; TileWallpaper = 0 }
+                    style = Desktop.CairoWallpaperStyle.Fill;
+                    break;
+                case "220": // Span { WallpaperStyle = 10; TileWallpaper = 0 }
+                    style = Desktop.CairoWallpaperStyle.Span;
+                    break;
+                case "20": // Stretched { WallpaperStyle = 2; TileWallpaper = 0 }
+                default:
+                    style = Desktop.CairoWallpaperStyle.Stretch;
+                    break;
+            }
+
+            txtWindowsBackgroundPath.Text = regWallpaper;
+            cboWindowsBackgroundStyle.Text = style.ToString();
+
+            #endregion
+            #region  cairoImageWallpaper
+            txtCairoBackgroundPath.Text = Settings.Instance.CairoBackgroundImagePath;
+            cboCairoBackgroundStyle.Text = Settings.Instance.CairoBackgroundImageStyle;
+            #endregion
+            #region  cairoVideoWallpaper
+            txtCairoVideoBackgroundPath.Text = Settings.Instance.CairoBackgroundVideoPath;
+            #endregion
+            #region  bingWallpaper
+            cboBingBackgroundStyle.Text = Settings.Instance.BingWallpaperStyle;
+            #endregion
+
+            var listBoxItems = cboDesktopBackgroundType.Items.Cast<ListBoxItem>().ToList();
+            var item = listBoxItems.FirstOrDefault(l => l.Name == Settings.Instance.DesktopBackgroundType);
+
+            cboDesktopBackgroundType.SelectedItem = item;
+        }
+
+        private void btnWindowsBackgroundFileBrowse_Click(object sender, RoutedEventArgs e)
+        {
+            string wallpaperPath;
+            if (ShowOpenFileDialog(GetImageFilter(), out wallpaperPath))
+            {
+                txtWindowsBackgroundPath.Text = wallpaperPath;
+            }
+        }
+
+        private void btnCairoBackgroundFileBrowse_Click(object sender, RoutedEventArgs e)
+        {
+            string wllpaperPath;
+            if (ShowOpenFileDialog(GetImageFilter(), out wllpaperPath))
+            {
+                txtCairoBackgroundPath.Text = wllpaperPath;
+            }
+        }
+
+        private void btnCairoVideoFileBrowse_Click(object sender, RoutedEventArgs e)
+        {
+            string wllpaperPath;
+            if (ShowOpenFileDialog(GetVideoFilter(), out wllpaperPath))
+            {
+                txtCairoVideoBackgroundPath.Text = wllpaperPath;
+            }
+        }
+
+        private bool ShowOpenFileDialog(string filter, out string filename)
+        {
+            bool result;
+
+            Microsoft.Win32.OpenFileDialog openFileDialog = new Microsoft.Win32.OpenFileDialog();
+            openFileDialog.Filter = filter;
+            if (openFileDialog.ShowDialog() == true)
```

Comment:
We need to put a try/catch in here-- Core versions of Windows (such as Hyper-V server) will fail to load the dialog because it by default gets upgraded to a new version which these OS don't support. In the catch, we disable the upgrade. See the method at `AppGrabberUI.xaml.cs` line 80. Maybe we should put this logic in a common place so we that we don't have to remember this?

Reply:
Made a Dialogs class in Common.